packages/@react-stately/layout/src/ListLayout.ts

```diff
@@ -431,41 +431,41 @@ export class ListLayout<T> extends Layout<Node<T>> implements KeyboardDelegate,
     return this.contentSize;
   }
 
-  getKeyAbove(key: Key): Key | null {
+  getKeyAbove(key: Key, itemFilter: (item: Node<T>) => boolean = item => item.type === 'item'): Key | null {
```

Comment:
Note the API change here to the getKey methods for ListLayout. This is so we can include sections as valid return keys from these methods ONLY during drag and drop. I'm a bit torn if modifying ListLayout is the best place to do this, but it does allow ListView's DnD to use the same code this way since they both provide a layout to useDroppableCollection.

Alternatives:
- create a Table DnD specific keyboard delegate 
  - Where would this live? Is it too specific? Opted not to do this for these reasons
- Modifying the Table/GridKeyboardDelegate `getKey` getters instead
  - Would need the delegate to be returned by useTable, changes felt odd. Code for this can be found before this [commit](https://github.com/adobe/react-spectrum/pull/4210/commits/fcf0c6e1dd908647ea83c17701835bacbc7615fb) aka in https://github.com/adobe/react-spectrum/pull/4210/commits/96f4921019619f27c56af427f65ee473670f0902